File: projects/smarthome/SmartHome-Version-4.0/app/src/main/java/com/sanjeev/stephan/murmu/ctsmartnetsmarthome/Show.java
package com.sanjeev.stephan.murmu.ctsmartnet.smarthome;

import android.content.Context;
import android.util.Log;
import android.widget.Toast;

public class Show
{

    private String location,tagName = "TAG";
    private String methodName = " ";
    private Context context;
    public Show(String location) {
        this.location = location;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    public void setLog(String msg)
    {
        Log.i(tagName,location.concat("=>").concat(methodName).concat(msg));
    }

    public void setMethod(String methodName)
    {
       this.methodName = "[" + methodName + "]";
    }

    public void setToast(String msg)
    {
        Toast.makeText(context,msg,Toast.LENGTH_SHORT).show();
    }

}

File: projects/smarthome/SmartHome-Version-Kotlin/app/src/main/java/com/sanjeev/stephan/murmu/smarthome/Show.kt
package com.sanjeev.stephan.murmu.smarthome

import android.app.Activity
import android.content.Context
import android.util.Log
import android.widget.Toast


class Show(private val context : Activity, private val location: String) {
    private val tagName = "TAG"
    private var methodName = " "

    fun setLog(msg: String) {
        Log.i(tagName, "$location=>$methodName$msg")
    }

    fun setToast(msg: String) {
        Toast.makeText(context, msg, Toast.LENGTH_SHORT).show()
    }

}